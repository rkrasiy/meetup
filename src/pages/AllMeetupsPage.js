
import MeetupItem from "../components/meetups/MeetupItem";
import classes from "./../components/meetups/MeetupList.module.css";

export default function AllMeetupsPage(props) {
  const { data , favorites } = props
  if (!data) return <p>Loading...</p>;
  return (
    <section>
      <h1>All Meetups</h1>
      <ul className={classes.list}>
        {
          data.map( item => {
            let isFavorite = false;
            if(favorites.includes(item.id)) {
              isFavorite = true;
            }
            return <MeetupItem item={item} key={item.id} isFavorite={isFavorite}/>
          })
        }
      </ul>
    </section>
  );
}
