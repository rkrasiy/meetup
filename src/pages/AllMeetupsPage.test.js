/* eslint-disable testing-library/no-debugging-utils */
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import AllMeetupsPage from "./AllMeetupsPage"

test("<AllMeetupsPage/> renders without crashing", () => {
  const wrapper = shallow(<AllMeetupsPage />);
  expect(wrapper.exists()).toBe(true);
});

describe('Check component AllMeetupsPage', ()=>{
  let wrapper = shallow(<AllMeetupsPage />);

  beforeEach(()=> {
    wrapper = shallow(<AllMeetupsPage />);
  })

  test('must be shown <AllMeetupsPage /> correctly', ()=> {
    expect(wrapper).toMatchSnapshot();
  })

  test('must shown Loading...', ()=> {
     const wrapper = shallow(<AllMeetupsPage />);
      const counterText = wrapper.find('p').text().trim();
      expect(counterText).toBe('Loading...');
  })

  test('must shown title: All Meetups', ()=> {
     const wrapper = shallow(<AllMeetupsPage data={[{'id':'a1','title':'title 1'}]} favorites={['a1']} />);
      const counterText = wrapper.find('h1').text().trim();
      expect(counterText).toBe('All Meetups');
  })
})