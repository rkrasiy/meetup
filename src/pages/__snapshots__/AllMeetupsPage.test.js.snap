// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check component AllMeetupsPage must be shown <AllMeetupsPage /> correctly 1`] = `ShallowWrapper {}`;

exports[`Check componentn AllMeetupsPage must be shown <AllMeetupsPage /> correctly 1`] = `ShallowWrapper {}`;
